src/common/api/hive.ts

```diff
@@ -599,6 +601,24 @@ export interface BlogEntry {
 export const getBlogEntries = (username: string, limit: number = dataLimit): Promise<BlogEntry[]> =>
   client.call("condenser_api", "get_blog_entries", [username, 0, limit]);
 
+export const getAccountVotesTrail = (
+  username: string,
+  start = -1,
+  limit = 20
+): Promise<BlogEntry[]> => {
+  let params = [username, start, limit, 1];
+  return client.call("condenser_api", "get_account_history", params).then((data) => {
```

Comment:
@igormuba you should use `getAccountHistory` api call which is already defined in `common/api/hive.ts` with filters. And you can create bitmask filter for fetching that way, you will only get `vote` operations and much less data to process. For example of `makeBitMaskFilter` check `common/store/transactions/index.ts` line ~105, similarly you can simply create filters for vote operations.

Reply:
@igormuba alright, I have fixed this myself.